src/main/java/com/amazonaws/services/kinesis/clientlibrary/lib/worker/Worker.java

```diff
@@ -1421,6 +1423,12 @@ public Worker build() {
                 throw new IllegalArgumentException(
                         "Kinesis Client Library configuration needs to be provided to build Worker");
             }
+            if (periodicShardSyncManager != null) {
+                if (leaseManager == null || shardSyncer == null || metricsFactory == null || leaderDecider == null) {
+
+                    throw new IllegalArgumentException("LeaseManager, ShardSyncer, MetricsFactory, and LeaderDecider must be provided if PeriodicShardSyncManager is provided");
+                }
+            }
```

Comment:
Why do we need this check? will the worker not start if not passed? will current implementations be affect with this?

Reply:
All of those elements are used in the construction of a PeriodicShardSyncManager, so I added that check to make sure no duplicate elements like the shardSyncer are created if a PeriodicShardSyncManager has already passed to the Worker. Current implementations should not be affected as this check only happens if a PeriodicShardSyncManager is passed to the Worker which did not happen prior to my changes.